### PR TITLE
Create UMIP-15

### PR DESCRIPTION
**Pull request by chrismaree**
Adds UMIP-15 flash loan mitigation DVM upgrade.

### CODE REVIEW

**Review by daywiss — COMMENTED** [review]
Looks good

File: UMIPs/UMIP-15 [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+## Motivation & Rationale
+
+The UMA voting mechanism uses a commit-reveal scheme, where votes are weighted in proportion to the token balance of the voters. The corresponding snapshot of the token distribution is taken at the start of each reveal phase, either through a dedicated function call, or automatically alongside the first revealed vote.
+
+However, it is possible for users to take out a flash loan of UMA tokens from a decentralized exchange, trigger the snapshot and then repay the loan in a single transaction. This would arguably give them undue influence over the result of the vote. Given the current state of the DeFi ecosystem, the flash-loan scenario is significantly simpler, more plausible, and arguably more concerning, so it warrants a dedicated mitigation.
+
+This UMIP proposes a solution that makes it imposable to utilize flash loans during voting, thereby enforcing that users hold UMA tokens if they wish to participate in governance.
```
Comment (nicholaspai) [inline UMIPs/UMIP-15:20]:
```suggestion
This UMIP proposes a solution that makes it imposable to utilize flash loans during voting, thereby enforcing that users hold UMA tokens beyond a single block if they wish to participate in governance.
```

File: UMIPs/UMIP-15 [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+This UMIP was originally motivated by an analysis conducted by the UMA engineering team that identified the issue in July 2020. The analysis can be found [here](https://docs.google.com/document/d/11ap5q2ga2OaVIV6MLxpRjzbLZTr0xraUIt2DdfcyzWI/edit?usp=sharing). The conclusion was that voting rounds with more than 2 votes are suspectable to a profitable flash loan. Subsequently, the flash loan issue was used during a DVM vote wherein uniswap LP tokens were used to to profitably vote without the flash borrower needing to hold any UMA tokens.
```
Comment (nicholaspai) [inline UMIPs/UMIP-15:22]: You should totally refercen this OZ forum thread that provides very valuable context in how we arrived at this conclusion: https://forum.openzeppelin.com/t/erc20snapshot-and-flash-loans-swaps-mints/3094/13

**Review by nicholaspai — DISMISSED** [review]

File: UMIPs/UMIP-15 (hunk 2 above)
Comment (chrismaree) [reply UMIPs/UMIP-15]: great idea and done!

**Review by mrice32 — APPROVED** [review]
LGTM!

**Review by daywiss — APPROVED** [review]

**Review by mrice32 — APPROVED** [review]

**Pull request merged**